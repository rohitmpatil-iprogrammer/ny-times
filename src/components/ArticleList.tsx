import React from "react";
import { Article } from "../services/api";
import CustomCard from "../common/CustomCard";
import { Helmet } from "react-helmet-async";

interface ArticleListProps {
  articles: Article[];
}

const ArticleList: React.FC<ArticleListProps> = ({ articles }) => {
  return (
    <>
      {articles.map((article) => {
        const imageSrc = article?.media?.[0]?.["media-metadata"]?.[2]?.url;
        return (
         <CustomCard id={article.id} abstract={article.abstract} title={article.title} imageSrc={imageSrc} />
        );
      })}
    </>
  );
};

export default ArticleList;
