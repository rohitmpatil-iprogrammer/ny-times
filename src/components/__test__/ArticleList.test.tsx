import { render } from "@testing-library/react";
import ArticleList from "../ArticleList";

test("renders ArticleList component correctly", () => {
  const { asFragment } = render(<ArticleList articles={[]} />);
  expect(asFragment()).toMatchSnapshot();
});
