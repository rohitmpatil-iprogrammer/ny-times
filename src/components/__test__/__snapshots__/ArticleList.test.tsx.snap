// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ArticleList component correctly 1`] = `<DocumentFragment />`;
